Makefile

```diff
@@ -17,28 +20,26 @@ testall: fennel
 	@printf "\nTesting lua 5.4:\n"; lua5.4 test/init.lua
 	@printf "\nTesting luajit:\n" ; luajit test/init.lua
 
-luacheck:
-	luacheck fennel.lua test/init.lua test/mangling.lua \
-		test/misc.lua test/quoting.lua
-
 count:
-	cloc fennel.lua
-	cloc --force-lang=lisp fennelview.fnl fennelfriend.fnl fennelbinary.fnl \
-		launcher.fnl
+	cloc --force-lang=lisp $(SRC) # core compiler
+	cloc --force-lang=lisp $(EXTRA_SRC) # libraries and launcher
 
-# For the time being, avoid chicken/egg situation thru the old Lua launcher.
-LAUNCHER=$(LUA) old_launcher.lua
+# Avoid chicken/egg situation using the old Lua launcher.
+LAUNCHER=$(LUA) old/launcher.lua --add-fennel-path src/?.fnl --globals "_G,_ENV"
 
 # Precompile fennel libraries
-%.lua: %.fnl fennel.lua
-	 $(LAUNCHER) --globals "" --compile $< > $@
+fennelview.lua: fennelview.fnl fennel.lua ; $(LAUNCHER) --compile $< > $@
+fennelfriend.lua: fennelfriend.fnl fennel.lua ; $(LAUNCHER) --compile $< > $@
 
 # All-in-one pure-lua script:
-fennel: launcher.fnl fennel.lua fennelview.lua fennelfriend.lua fennelbinary.fnl
+fennel: launcher.fnl $(SRC) fennelview.lua fennelfriend.lua fennelbinary.fnl
 	echo "#!/usr/bin/env $(LUA)" > $@
-	$(LAUNCHER) --globals "" --require-as-include --metadata --compile $< >> $@
+	$(LAUNCHER) --no-metadata --require-as-include --compile $< >> $@
```

Comment:
Why the change to `--no-metadata`? Was this to get around the problem with circular requires?

My only issue there is that it's nice to be able to use `doc` on fennelview from within the standalone launcher.